### PR TITLE
add jenkins test script

### PR DESCRIPTION
**Pull request by slefrancois**

### CODE REVIEW

File: .jenkins_tests [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Get environment from worker, necessary for CUDA
+source ~/.bashrc
+
+# Test theano CPU and old GPU backend sandbox.cuda
+THEANO_PARAM="theano --with-timer --timer-top-n 10"
+FLAGS="mode=FAST_RUN,init_gpu_device=gpu"
+THEANO_FLAGS=${FLAGS} bin/theano-nose ${THEANO_PARAM}
+
+# Build libgpuarray and test gpuarray backend
+GPUARRAY=none
```
Comment (nouiz) [inline .jenkins_tests]: not needed in jenkins.

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

File: .jenkins/jenkins_pretest.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+source ~/.bashrc
+
+# Test flake8
+echo "===== Testing flake8"
+bin/theano-nose theano/tests/test_flake8.py
+# Test documentation
+echo "===== Testing documentation"
+python doc/scripts/docgen.py --nopdf --check
+python doc/scripts/docgen.py --test --check
```
Comment (nouiz) [inline .jenkins/jenkins_pretest.sh]: Add before this line:


+echo "===== Testing doctest"

File: .jenkins/jenkins_pretest.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# Script for Jenkins continuous integration pre-testing
+
+source ~/.bashrc
+
+# Test flake8
+echo "===== Testing flake8"
+bin/theano-nose theano/tests/test_flake8.py
+# Test documentation
+echo "===== Testing documentation"
+python doc/scripts/docgen.py --nopdf --check
```
Comment (nouiz) [inline .jenkins/jenkins_pretest.sh]: python doc/scripts/docgen.py --nopdf --check || exit 1

idem for others.

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
Github PR to test.

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Comment by slefrancois** [comment]
jenkins test this please

**Pull request merged**